### PR TITLE
docs(rfc): button group

### PR DESCRIPTION
**Pull request by gergelyke**

### CODE REVIEW

File: rfcs/button-group-component.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+* `disabled: boolean` - Optional:
+  Defines if the button group disabled.
+* `size: 'default' | 'compact'`
+  Defines the size of a button
+* `onChange: func` - Optinal:
+  Called when a toggleable Button is selected
+* `onClick: func` - Optional:
+  Called with the click event
+* `checked: numer | Array<number>` - Optional:
+  Index or array of indices of the selected
```
Comment (nadiia) [inline rfcs/button-group-component.md]: What is this prop for? Seems to be not related.

File: rfcs/button-group-component.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## Exports
+
+* `ButtonGroup`
+
+## `ButtonGroup` API
+
+* `children: (props: Props) => React.Node` - Required
+  This can be an SVG icon as well.
+* `ariaLabel: string` - Required:
```
Comment (abmai) [inline rfcs/button-group-component.md]: Do we need to enforce aria labels?

File: rfcs/button-group-component.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  Accessible label
+* `disabled: boolean` - Optional:
+  Defines if the button group disabled.
+* `size: 'default' | 'compact'`
```
Comment (abmai) [inline rfcs/button-group-component.md]: How do you see this prop working with the underlying `Button`, via this composition approach?

File: rfcs/button-group-component.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Button} from 'baseui/button';
+import {ButtonGroup} from 'baseui/button-group';
+
+export default () => {
+  return
+    <ButtonGroup>
```
Comment (blackswanny) [inline rfcs/button-group-component.md]: button group can change styles around button, so we need a root component to be exposed for that

File: rfcs/button-group-component.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+## `ButtonGroup` API
+
+* `children: (props: Props) => React.Node` - Required
+  This can be an SVG icon as well.
+* `ariaLabel: string` - Optional:
+  Accessible label
+* `disabled: boolean` - Optional:
+  Defines if the button group disabled.
+* `size: 'default' | 'compact'`
```
Comment (abmai) [inline rfcs/button-group-component.md]: Not sure why the other conversation got collapsed :(. But Do you think it's worth providing all the `Button` config API here and just say that it'll automatically be applied to all children `Button`? i.e. `KIND`, `SIZE`, `SHAPE`

Comment (gergelyke) [reply rfcs/button-group-component.md]: I think that makes sense - thoughts @nadiia @schnerd @blackswanny ?

**Comment by gergelyke** [comment]
> Will the same onClick and onChange from the ButtonGroup props be attached to every child button?

not sure - can we think of any use-cases when we would want a separate event handler for the group and the button instances? 🤔 if not, probably it worth passing them in

File: rfcs/button-group-component.md (hunk 5 above)
Comment (blackswanny) [reply rfcs/button-group-component.md]: Yes, the same happens in other components. Props fall down to nested components.
For instance, Select sends props to each Option

File: rfcs/button-group-component.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  Defines the size of a button
+* `onChange: func` - Optinal:
+  Called when a toggleable Button is selected
+* `onClick: func` - Optional:
+  Called with the click event
```
Comment (schnerd) [inline rfcs/button-group-component.md]: So just to clarify, is this component meant to behave almost like a radio group? Or is it just meant to visually group the buttons together?

If just the latter, then maybe we don't need `onChange` and `onClick` handlers here. I'm trying to figure out what I would do with those as a customer – seems like it makes more sense to just attach onClick to each individual buttons.

But if it's the former (more like a radio group, where a button can be "selected"), then I think we may want to align more closely with the radio group API. This is what antd does: https://codesandbox.io/s/6xoqyy57o3

Which one were you thinking?

cc @nadiia @abmai @blackswanny

File: rfcs/button-group-component.md (hunk 5 above)
Comment (schnerd) [reply rfcs/button-group-component.md]: I like the idea of propagating the `kind`, `size`, `shape`, and related props down to buttons, though we might want to explicitly pass them rather than just passing `{...restProps}` down to each button.

File: rfcs/button-group-component.md (hunk 6 above)
Comment (gergelyke) [reply rfcs/button-group-component.md]: oh, I see - [mineral-ui](https://mineral-ui.com/components/button-group) does something very similar.

I think based on this discussion, we should follow up with design to see what behavior they had in mind - does that sound reasonable?

Comment (schnerd) [reply rfcs/button-group-component.md]: Yeah that's probably a good idea.

File: rfcs/button-group-component.md (hunk 5 above)
Comment (nadiia) [reply rfcs/button-group-component.md]: Totally makes sense since we already moving the `size` up

File: rfcs/button-group-component.md (hunk 6 above)
Comment (nadiia) [reply rfcs/button-group-component.md]: Yes, let's extend the question to the design team. But I'd assume every button would need an individual click or change handler.

File: rfcs/button-group-component.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+* `mode: string` - Optional:
+  Use the `mode` prop to render toggleable Buttons:
+  * the value `radio` will cause Buttons to behave like radio buttons,
+  * the value `checkbox` will cause Buttons to behave like checkboxes.
+* `defaultChecked: number | Array<number>`
+  Index or array of indices of the selected Button(s).
+  Primarily for use with uncontrolled components with a `mode` prop defined.
+* `checked: number | Array<number>` - Optional
+  Index or array of indices of the selected Button(s).
+  Primarily for use with controlled components with a `mode` prop defined.
```
Comment (schnerd) [inline rfcs/button-group-component.md]: So the `defaultChecked`/`checked` props sort of allow you to use this component as both a controlled (stateless) and uncontrolled (stateful) component. If we want to be more consistent with the rest of our BaseUI components, it may be preferable to have a `StatefulButtonGroup` with an `initialState prop` and a `ButtonGroup` with the `checked` prop.

File: rfcs/button-group-component.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
+# Button Group Component
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Button} from 'baseui/button';
```
[the remark this answers is not in the document]
Comment (gergelyke) [reply rfcs/button-group-component.md]: what do you think @nadiia @abmai @blackswanny?

File: rfcs/button-group-component.md (hunk 7 above)
Comment (gergelyke) [reply rfcs/button-group-component.md]: ah, you are right, I'll update it

File: rfcs/button-group-component.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (blackswanny) [reply rfcs/button-group-component.md]: The task of integrating already existing components (radio\checkbox) becomes non-trivial.
In fact we need to repeat the same interface we have for them. I think, just visual appearance shouldn't lead us to make things more complex.
I vote for not adding radio\checkbox modes in here, but just use `override` there or add some default styles in those components for these cases.

File: rfcs/button-group-component.md (hunk 4 above)
Comment (blackswanny) [reply rfcs/button-group-component.md]: did we add extra root element to be able to change padding\margin, etc around buttons?

File: rfcs/button-group-component.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (abmai) [reply rfcs/button-group-component.md]: Agreed with David's second option as well, new `Button` that's meant to be used inside this `ButtonGroup` but built on top of `Button` and takes the same overrides.

File: rfcs/button-group-component.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+* `onChange: func` - Optinal:
+  Called when a toggleable Button is selected
+* `onClick: func` - Optional:
+  Called with the click event
+
+## `StatefulButtonGroup` API
+
+Extends the `ButtonGroup` API with this additional props:
+
+* `initalState: number | Array<number>`
```
Comment (schnerd) [inline rfcs/button-group-component.md]: Most of our initialState props are objects, meant to represent the normal state object. Maybe we want to do that here too? `initialState: {checked: number | Array<number>}`
